### PR TITLE
Enable amd64 arch in sawtooth repository links

### PR DESCRIPTION
**Pull request by askmish**
Enabled only amd64 architecture for sawtooth: stable and nightly repo used in add-apt-repository

### CODE REVIEW

**Comment by askmish** [comment]
@vaporos Here's what I get on my ubuntu 16.04.4 LTS x64 machine:
`N: Skipping acquire of configured file 'universe/binary-i386/Packages' as repository 'http://repo.sawtooth.me/ubuntu/1.0/stable xenial InRelease' doesn't support architecture 'i386'`

This is the simplest, standard fix to limit arch to only 64-bit.

**Comment by rbuysse** [comment]
How can I reproduce this? I don't experience this behavior on a fresh ubuntu machine.

**Comment by askmish** [comment]
I followed these steps:
$ sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8AA7AF1F1091A5FD
$ sudo add-apt-repository 'deb http://repo.sawtooth.me/ubuntu/1.0/stable xenial universe'
$ sudo apt-get update

I can consistently reproduce this issue on all ubuntu machines I have.

**Comment by askmish** [comment]
ping

**Comment by vaporos** [comment]
An example where this is done outside our project:

https://docs.docker.com/install/linux/docker-ce/ubuntu/#set-up-the-repository

Do we know of others?

**Comment by vaporos** [comment]
I think we have consensus that this is an okay approach.  However, I would like us to do this then in all places which reference these repos and not just the app dev guide.  (Docker images for example, etc.)

**Comment by askmish** [comment]
Ok. we'll put up a separate PR for this.

**Comment by vaporos** [comment]
Let's do it in this PR please.

**Comment by askmish** [comment]
@vaporos The purpose of this PR would change completely. Is that ok.

File: docker/sawtooth-validator-mounted [hunk 1]
```diff
@@ -96,7 +96,7 @@ RUN mkdir -p /project/sawtooth-core/ \
 ENV PATH=$PATH:/project/sawtooth-core/bin
 
 RUN curl -sL https://repos.influxdata.com/influxdb.key | apt-key add -
-RUN echo deb https://repos.influxdata.com/ubuntu xenial stable > /etc/apt/sources.list.d/influxdb.list
+RUN echo deb [arch=amd64] https://repos.influxdata.com/ubuntu xenial stable > /etc/apt/sources.list.d/influxdb.list
```
Comment (askmish) [inline docker/sawtooth-validator-mounted]: Should we update this as well?

Comment (vaporos) [reply docker/sawtooth-validator-mounted]: No, it is not our repo.

Comment (askmish) [reply docker/sawtooth-validator-mounted]: ok fixed it

**Comment by askmish** [comment]
The CI build has verfied changes in docker/ci dirs. For changes in docs, manual review needed.

File: ci/sawtooth-build-debs [hunk 2]
```diff
@@ -29,7 +29,7 @@
 FROM ubuntu:xenial
 
 # Add additional xenial repos
-RUN sh -c "echo deb http://archive.ubuntu.com/ubuntu xenial-backports main restricted universe multiverse >> /etc/apt/sources.list" && \
+RUN sh -c "echo deb [arch=amd64] http://archive.ubuntu.com/ubuntu xenial-backports main restricted universe multiverse >> /etc/apt/sources.list" && \
```
Comment (vaporos) [inline ci/sawtooth-build-debs]: This isn't a repo.sawtooth.me repo, so shouldn't need this change.

File: ci/sawtooth-publish-js-sdk [hunk 3]
```diff
@@ -38,7 +38,7 @@
 FROM ubuntu:xenial
 
 # Add additional xenial repos
-RUN sh -c "echo deb http://archive.ubuntu.com/ubuntu xenial-backports main restricted universe multiverse >> /etc/apt/sources.list" && \
+RUN sh -c "echo deb [arch=amd64] http://archive.ubuntu.com/ubuntu xenial-backports main restricted universe multiverse >> /etc/apt/sources.list" && \
```
Comment (vaporos) [inline ci/sawtooth-publish-js-sdk]: Not a repo.sawtooth.me URL, so shouldn't need this change.

File: docker/sawtooth-dev-go [hunk 4]
```diff
@@ -29,8 +29,8 @@ FROM ubuntu:xenial
 
 LABEL "install-type"="mounted"
 
-RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sources.list \
- && echo "deb http://archive.ubuntu.com/ubuntu xenial-backports universe" >> /etc/apt/sources.list \
+RUN echo "deb [arch=amd64] http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sources.list \
+ && echo "deb [arch=amd64] http://archive.ubuntu.com/ubuntu xenial-backports universe" >> /etc/apt/sources.list \
```
Comment (vaporos) [inline docker/sawtooth-dev-go]: In the 2nd line here, it's not a repo.sawtooth.me repo.

**Comment by rberg2** [comment]
To see this warning you need to have the i386 architecture enabled on a multi-arch system, I believe amd64 Ubuntu does by default. To check which archs are enabled ```dpkg --print-foreign-architectures``` I have tested this change in a VM and it does silence the i386 warning on multi-arch systems.

**Pull request merged**